pkg/controller/controller.go

```diff
@@ -19,56 +33,595 @@ type Controller struct {
 	Client tcs.ExtensionInterface
 	// sync time to sync the list.
 	SyncPeriod time.Duration
+	// image of sidecar container
+	Image string
 }
 
-func New(c *rest.Config) *Controller {
+func New(c *rest.Config, image string) *Controller {
 	return &Controller{
 		Client:     tcs.NewExtensionsForConfigOrDie(c),
 		SyncPeriod: time.Minute * 2,
+		Image:      image,
 	}
 }
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
 // Blocks caller. Intended to be called as a Go routine.
 func (w *Controller) RunAndHold() {
 	lw := &cache.ListWatch{
 		ListFunc: func(opts api.ListOptions) (runtime.Object, error) {
-			return w.Client.Certificate(api.NamespaceAll).List(api.ListOptions{})
+			return w.Client.Backup(api.NamespaceAll).List(api.ListOptions{})
 		},
 		WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
-			return w.Client.Certificate(api.NamespaceAll).Watch(api.ListOptions{})
+			return w.Client.Backup(api.NamespaceAll).Watch(api.ListOptions{})
 		},
 	}
 	_, controller := cache.NewInformer(lw,
-		&tapi.Certificate{},
+		&rapi.Backup{},
 		w.SyncPeriod,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				glog.Infoln("Got one added tpr", obj.(*tapi.Certificate))
-				w.doStuff(obj.(*tapi.Certificate))
+				glog.Infoln("Got one added bacup obejct", obj.(*rapi.Backup))
```

Comment:
You must ensure the type of obj is Backup.
```
if backup, ok := obj.(*rapi.Backup); ok {
  do things...
}
```
otherwise things coming next are going to be paniced.